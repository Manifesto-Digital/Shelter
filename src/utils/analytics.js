export const sendEvent = data => {
  window.dataLayer.push(data);
};
