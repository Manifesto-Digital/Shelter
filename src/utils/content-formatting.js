export const addMarksToContent = content => {
  const markup = content
    .map(block => {
      const { marks, value } = block
      // As there may be multiple marks per piece of content, we need to
      // store the opening and closing tags and then wrap the content
      if (marks && marks.length > 0) {
        let openingMarks = ''
        let closingMarks = ''

        marks.forEach(mark => {
          if (mark.type === 'underline') {
            openingMarks = `<span className="underline">${openingMarks}`
            closingMarks = `${closingMarks}</span>`
          }
          if (mark.type === 'italic') {
            openingMarks = `<em>${openingMarks}`
            closingMarks = `${closingMarks}</em>`
          }
          if (mark.type === 'bold') {
            openingMarks = `<strong>${openingMarks}`
            closingMarks = `${closingMarks}</strong>`
          }
        })
        return `${openingMarks}${value}${closingMarks}`
      }
      return value
    })
    .join('')

  return markup
}
