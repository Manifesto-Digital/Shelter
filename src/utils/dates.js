import dayjs from 'dayjs';

// Format date as string in required format
// Formatting options: https://bit.ly/2R9uVsh
export const dateAsString = (date, format) =>
  dayjs(date)
    .format(format)
    .toString();

export const getWeekDays = () => {
  // Base date is a Monday, this ensures we get the day lsit in order
  const baseDate = new Date(Date.UTC(2019, 0, 7));
  const weekDays = [];
  for (let i = 0; i < 7; i += 1) {
    weekDays.push(baseDate.toLocaleDateString('en-GB', { weekday: 'long' }));
    baseDate.setDate(baseDate.getDate() + 1);
  }
  return weekDays;
};

export const dayOfTheWeek = () => {
  const daysArray = getWeekDays();
  const today = new Date();

  return daysArray[today.getDay() - 1];
};
