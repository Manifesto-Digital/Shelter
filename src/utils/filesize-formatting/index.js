/**
 * Formats integer to human readable filesize
 *
 * @param {int} bytes
 * @param {int} decimalPoint
 */
export const formatFilesize = (bytes, decimalPoint) => {
  if (bytes === undefined) throw new Error('Bytes are required');
  if (bytes < 0) throw new Error('Negative bytes not allowed');
  if (bytes >= 10 ** 27)
    throw new Error(`Bytes too large, must be < 10 ** 27 (${10 ** 27})`);

  if (bytes === 0) return '0 Bytes';

  const k = 1000;
  const dm = decimalPoint || 2;
  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
  const i = Math.floor(Math.log(bytes) / Math.log(k));

  const sizeToReturn = `${parseFloat((bytes / k ** i).toFixed(dm))} ${
    sizes[i]
  }`;
  return sizeToReturn;
};
