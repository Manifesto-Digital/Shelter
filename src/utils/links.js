export const getInternalLink = slug => {
  const baseURL = window.location.origin
  return `${baseURL}/${slug}`
}
