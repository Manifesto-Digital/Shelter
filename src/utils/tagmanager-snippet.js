const tagManagerSnippet = () =>
  (function(w, d, s, l, i) {
    w[l] = w[l] || [];
    w[l].push({
      'gtm.start': new Date().getTime(),
      event: 'gtm.js',
    });
    const f = d.getElementsByTagName(s)[0];
    const j = d.createElement(s);
    const dl = l != 'dataLayer' ? `&l=${l}` : ''; // eslint-disable-line eqeqeq
    j.async = true;
    j.src = `https://www.googletagmanager.com/gtm.js?id=${i}${dl}`;
    f.parentNode.insertBefore(j, f);
  })(window, document, 'script', 'dataLayer', 'GTM-WWSW2HH');

export default tagManagerSnippet;
