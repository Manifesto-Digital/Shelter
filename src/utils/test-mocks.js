// Whenever LinkHandler is used, when mounted in enzyme the rendered HTML
// includes a lowercase "l" in the gatsby Link tag. Which causes a warning
// This suppresses that warning for a whole test file
export const hidePascalCaseWarning = () => {
  beforeEach(() => {
    console.error = jest.fn(error => {
      if (
        !error.includes(
          'using incorrect casing. Use PascalCase for React components, or lowercase for HTML elements'
        )
      ) {
        throw new Error(error);
      }
    });
  });

  afterEach(() => {
    console.error.mockClear();
  });
};
