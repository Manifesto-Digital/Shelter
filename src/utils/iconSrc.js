import SpeechBubble from '../assets/svg/icons/comment-light.svg';
import Envelope from '../assets/svg/icons/envelope-light.svg';
import Download from '../assets/svg/icons/download-light.svg';
import Printer from '../assets/svg/icons/print-light.svg';
import OpenBook from '../assets/svg/icons/book-open-light.svg';
import Exclamation from '../assets/svg/icons/exclamation-circle-light.svg';
import Telephone from '../assets/svg/icons/phone-light.svg';
import MapMarker from '../assets/svg/icons/map-marker-alt-light.svg';
import ChippedTimeResults from '../assets/svg/icons/challenge-event/chipped-time-results.svg';
import FinishersMedal from '../assets/svg/icons/challenge-event/finishers-medal.svg';
import FreeBagTransfer from '../assets/svg/icons/challenge-event/free-bag-transfer.svg';
import FreeRacePhotos from '../assets/svg/icons/challenge-event/free-race-photos.svg';
import GoodieBags from '../assets/svg/icons/challenge-event/goodie-bags.svg';
import PostRaceCelebration from '../assets/svg/icons/challenge-event/post-race-celebration.svg';
import SportsMassage from '../assets/svg/icons/challenge-event/sports-massage.svg';
import TechnicalRunningTop from '../assets/svg/icons/challenge-event/technical-running-top.svg';
import ClipboardChecklist from '../assets/svg/icons/clipboard-checklist-light.svg';
import FundingGoalMeter from '../assets/svg/icons/funding-goal-meter-light.svg';
import Lightbulb from '../assets/svg/icons/lightbulb-light.svg';
import MapUnfolded from '../assets/svg/icons/map-unfolded-light.svg';

import { consistentString } from './content-formatting';

const iconList = {
  'speech-bubble': SpeechBubble,
  envelope: Envelope,
  download: Download,
  printer: Printer,
  'open-book': OpenBook,
  'exclamation-mark': Exclamation,
  telephone: Telephone,
  'map-marker': MapMarker,
  'chipped-time-results': ChippedTimeResults,
  'finishers-medal': FinishersMedal,
  'free-bag-transfer': FreeBagTransfer,
  'free-race-photos': FreeRacePhotos,
  'goodie-bags': GoodieBags,
  'post-race-celebration': PostRaceCelebration,
  'sports-massage': SportsMassage,
  'technical-running-top': TechnicalRunningTop,
  'clipboard-checklist': ClipboardChecklist,
  'funding-goal-meter': FundingGoalMeter,
  lightbulb: Lightbulb,
  'map-unfolded': MapUnfolded,
  'no-icon': null,
  default: null,
};

// returns the src of an svg icon from the above icon list
const iconSrc = icon => {
  const iconKey = consistentString(icon);

  if (iconList[iconKey] === undefined) {
    throw new Error(
      'No matching icon was found. Is it set as an option in the CMS? Has it been imported and included in the object above?'
    );
  }

  const chosenIcon = iconList[iconKey] || null;

  if (!chosenIcon) return null;

  return chosenIcon;
};

export default iconSrc;
