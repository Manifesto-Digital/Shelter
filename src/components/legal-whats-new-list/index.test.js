import React from 'react';
import { Link } from 'gatsby';
import { snapshotComponent, mountWithTheme } from 'test-helpers';
import LinkList from '.';
import { hidePascalCaseWarning } from '../../utils/test-mocks';

const mockData = {
  featuredItem: {
    title: 'Featured legal page',
    slug: 'featured-legal-page',
    lastAmended: '2019-03-26T00:00+00:00',
    pageInformation: {
      shortDescription: {
        shortDescription:
          'Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Nullam nulla eros, ultricies sit amet, nonummy id, imperdiet feugiat, pede. In turpis. short desc',
      },
    },
  },
  items: [
    {
      title: 'Legal Page 1',
      slug: 'legal-page-1',
      lastAmended: '2019-03-26T00:00+00:00',
      pageInformation: {
        shortDescription: {
          shortDescription:
            'Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Nullam nulla eros, ultricies sit amet, nonummy id, imperdiet feugiat, pede. In turpis. short desc',
        },
      },
    },
    {
      title: 'Legal page 2',
      slug: 'legal-page-2',
      lastAmended: '2019-03-25T00:00+00:00',
      pageInformation: {
        shortDescription: {
          shortDescription:
            'Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Nullam nulla eros, ultricies sit amet, nonummy id, imperdiet feugiat, pede. In turpis. short desc',
        },
      },
    },
  ],
};

it('renders correctly', () => {
  snapshotComponent(<LinkList {...mockData} />);
});

hidePascalCaseWarning();

it('renders a the featured item and the array of items with links', () => {
  const wrapper = mountWithTheme(<LinkList {...mockData} />);

  const items = [mockData.featuredItem, ...mockData.items];

  expect(wrapper.find('li').length).toEqual(items.length);

  wrapper.find(Link).forEach((link, i) => {
    expect(link.prop('to')).toContain(items[i].slug);
  });
});
