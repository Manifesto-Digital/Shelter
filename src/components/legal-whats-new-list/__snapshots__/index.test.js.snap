// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  padding-top: 40px;
}

.c1 {
  list-style-type: none;
  padding: 0;
  margin: 0;
}

.c2 {
  margin-bottom: 40px;
  padding-left: 20px;
  border-left: 4px solid #ea232d;
}

.c5 {
  margin-bottom: 40px;
  padding-left: 20px;
  border-left: 4px solid #999;
}

.c3 {
  margin-bottom: 10px;
}

.c4 {
  margin-bottom: 40px;
}

<section
  className="c0"
>
  <ul
    className="c1"
  >
    <li
      className="c2"
    >
      <h3
        className="c3"
      >
        Featured legal page
      </h3>
      <p
        className="c4"
      >
        26 March 2019
      </p>
      <p>
        Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Nullam nulla eros, ultricies sit amet, nonummy id, imperdiet feugiat, pede. In turpis. short desc
      </p>
      <Link
        to="/featured-legal-page"
      >
        View amended article
      </Link>
    </li>
    <li
      className="c5"
    >
      <h3
        className="c3"
      >
        Legal Page 1
      </h3>
      <p
        className="c4"
      >
        26 March 2019
      </p>
      <p>
        Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Nullam nulla eros, ultricies sit amet, nonummy id, imperdiet feugiat, pede. In turpis. short desc
      </p>
      <Link
        to="/legal-page-1"
      >
        View amended article
      </Link>
    </li>
    <li
      className="c5"
    >
      <h3
        className="c3"
      >
        Legal page 2
      </h3>
      <p
        className="c4"
      >
        25 March 2019
      </p>
      <p>
        Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Nullam nulla eros, ultricies sit amet, nonummy id, imperdiet feugiat, pede. In turpis. short desc
      </p>
      <Link
        to="/legal-page-2"
      >
        View amended article
      </Link>
    </li>
  </ul>
</section>
`;
